includes/admin/class-sensei-learner-management.php

```diff
@@ -272,12 +273,56 @@ public function load_data_object( $name = '', $data = 0, $optional_data = null )
 	 * @access public
 	 */
 	public function learners_page() {
-		$type = isset( $_GET['view'] ) ? esc_html( $_GET['view'] ) : false;
-		if ( $this->bulk_actions_controller->get_view() === $type ) {
-			$this->bulk_actions_controller->learner_admin_page();
-			return;
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
+		if ( ! empty( $_GET['course_id'] ) ) {
+			$this->output_course_page();
+		} else {
+			$this->output_main_page();
 		}
 
+	}
```

Comment:
I've decided to move the menu page rendering logic to the main class. We basically have two different tables - one for all students (with bulk actions and filters) and one in the context of a course.

Another option would be to create a new class but this might end up in a rabbit hole of refactoring. Let me know if you have other suggestions.

Reply:
Looks good to me